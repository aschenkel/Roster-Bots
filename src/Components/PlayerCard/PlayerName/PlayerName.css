.content{
  margin: 16px
}
.name{
    font-size: 19px;
    font-weight: bold;
    display: block;
    width: 370px;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    color: #1C0113;
    margin: -8px 0 0 0 
}

.name-tag{
    font-size: 14px;
    display: block;
    text-shadow: 0px 1px 0px rgba(255,255,255,.3), 0px -1px 0px rgba(0,0,0,.7);
    color:gray
}