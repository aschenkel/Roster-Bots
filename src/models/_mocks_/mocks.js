import {numberOfPlayersPerTeam,maxScoreAllowed, numberOfStarters} from '../../config/rules'


export const createMockPlayer = (score) =>{
    var halfScore = score/2
    return {
        name:mockNameCreator(),
        speed: Math.round(halfScore),
        strenght:Math.floor(halfScore),
        score
    }
}

export const mockNameCreator = () => {
    return '081813ad-c751-40e6-a12b-04f61968607b'
}

export const createMockRoster = () =>{
    var mockPlayers = []
    for (var i = 0; i < numberOfPlayersPerTeam; i++) {
      var properScore = maxScoreAllowed - i;
      mockPlayers.push(createMockPlayer(properScore))
    }
    return mockPlayers
}


