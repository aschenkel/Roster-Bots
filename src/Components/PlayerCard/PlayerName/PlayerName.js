import React from 'react';
import PropTypes from 'prop-types';
import './PlayerName.css';

const PlayerName = (props) => {
    return (
      <div className="content">
          <p className="name-tag">name</p>
          <p className="name">{props.name}</p>
        </div>
    );
}

PlayerName.propTypes = {
  name: PropTypes.string.isRequired
};


export default PlayerName;