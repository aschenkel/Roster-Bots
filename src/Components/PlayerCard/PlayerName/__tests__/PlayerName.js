import React from 'react';
import PlayerName from '../PlayerName';
import renderer from 'react-test-renderer';
import {mockNameCreator} from '../../../../models/_mocks_/mocks'

it('should render a playerÂ´s name', () => {
    var mockName = mockNameCreator()
    const tree = renderer.create(
        <PlayerName name={mockName}/>
        ).toJSON();
    expect(tree).toMatchSnapshot();
});

