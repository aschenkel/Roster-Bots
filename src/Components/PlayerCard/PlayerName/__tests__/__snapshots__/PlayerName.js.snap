// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a playerÂ´s name 1`] = `
<div
  className="content"
>
  <p
    className="name-tag"
  >
    name
  </p>
  <p
    className="name"
  >
    081813ad-c751-40e6-a12b-04f61968607b
  </p>
</div>
`;
